src/BaseClient.php

```diff
@@ -108,53 +136,137 @@ public function getToken(): ?array
      * @throws Exception when something unexpected went wrong.
      */
     public function authenticate(string $clientId, string $clientSecret): void
```

Comment:
Ik denk dat het misschien netjes is om deze methode te deprecaten, en een alias te maken voor (bijvoorbeeld) `authorizeByClientCredentials`. Dan is het namelijk in lijn met `authenticateByAuthorizationCode`.